pkg/controllers/management/node/controller.go

```diff
@@ -211,6 +210,11 @@ func (m *Lifecycle) getNodeTemplate(nodeTemplateName string) (*v3.NodeTemplate,
 	return m.nodeTemplateClient.GetNamespaced(ns, n, metav1.GetOptions{})
 }
 
+func (m *Lifecycle) getNodePool(nodePoolName string) (*v3.NodePool, error) {
+	ns, p := ref.Parse(nodePoolName)
+	return m.nodePoolLister.Get(ns, p)
```

Comment:
It looks too trivial to be a standalone function which is only used once.

Reply:
I would prefer this function stays with `getNodeTemplate`.